build.gradle

```diff
@@ -0,0 +1,171 @@
+    useTestNG()
+    if (project.hasProperty('test-name')) {
+        include '**/' + project.property('test-name') + '.class'
+    } else {
+        include '**/OrderTest.class'
+    }
+    testLogging {
+        events "PASSED", "FAILED", "SKIPPED"
+    }
+}
+
+tasks.register('on-prem-suite', Test) {
+    useTestNG()
+    exclude '**/OfferTest.class'
+    testLogging {
+        events "PASSED", "FAILED", "SKIPPED"
+    }
+}
+
+tasks.register('docker', Test) {
+    useTestNG()
+    if (project.hasProperty('test-name')) {
+        include '**/' + project.property('test-name') + '.class'
+    } else {
+        include '**/OrderTest.class'
+    }
+    systemProperty "environment", "docker"
+    testLogging {
+        events "PASSED", "FAILED", "SKIPPED"
+    }
+}
+
+tasks.register('docker-parallel', Test) {
+    useTestNG() {
+        parallel project.property('testngParallel')
+        threadCount Integer.parseInt(project.property('testngThreadCount'))
+    }
+    exclude '**/OfferTest.class'
+    systemProperty "environment", "docker"
+    testLogging {
+        events "PASSED", "FAILED", "SKIPPED"
+    }
+}
+
+tasks.register('bstack-single', Test) {
+    useTestNG() {
+        listeners.add("com.browserstack.test.utils.BrowserstackTestStatusListener")
+    }
+    if (project.hasProperty('test-name')) {
+        include '**/' + project.property('test-name') + '.class'
+    } else {
+        include '**/OrderTest.class'
+    }
+    systemProperty "environment", "remote"
+    systemProperty "test", "single"
+    testLogging {
+        events "PASSED", "FAILED", "SKIPPED"
+    }
+}
+
+tasks.register('bstack-local', Test) {
+    useTestNG() {
+        listeners.add("com.browserstack.test.utils.BrowserstackTestStatusListener")
+    }
+    if (project.hasProperty('test-name')) {
+        include '**/' + project.property('test-name') + '.class'
+    } else {
+        include '**/OrderTest.class'
+    }
+    systemProperty "environment", "remote"
+    systemProperty "test", "local"
+    testLogging {
+        events "PASSED", "FAILED", "SKIPPED"
+    }
+}
+
+tasks.register('bstack-local-parallel', Test) {
+    useTestNG() {
+        parallel project.property('testngParallel')
+        threadCount Integer.parseInt(project.property('testngThreadCount'))
+        listeners.add("com.browserstack.test.utils.BrowserstackTestStatusListener")
+    }
+    exclude '**/OfferTest.class'
+    systemProperty "environment", "remote"
+    systemProperty "test", "local"
+    testLogging {
+        events "PASSED", "FAILED", "SKIPPED"
+    }
+}
+
+tasks.register('bstack-local-parallel-browsers', Test) {
+    useTestNG() {
+        suites 'src/test/resources/conf/runners/testng-browserstack-parallel.xml'
+        parallel project.property('testngParallel')
+        threadCount Integer.parseInt(project.property('testngThreadCount'))
+        listeners.add("com.browserstack.test.utils.BrowserstackTestStatusListener")
+    }
+    exclude '**/OfferTest.class'
```

Comment:
since OfferTest is working now as per your Slack, please remove all excludes from gradle and pom files. Do an end to end test of the execution profiles. Please update here once done.